### PR TITLE
Introduced pipeline-based structure for docs and pipelines folder

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

**Comment by AlekseySh** [comment]
Changelog:
- rework folders in pipelines (existing pipelines have been split to extractor and postprocessor)
- `RetrievalModule` -> `ExtractorModule`
- Docs strings for pipeline entrypoints
-

File: oml/lightning/pipelines/train.py [hunk 1]
```diff
@@ -59,10 +59,11 @@ def get_retrieval_loaders(cfg: TCfg) -> Tuple[DataLoader, DataLoader]:
 
 def extractor_training_pipeline(cfg: TCfg) -> None:
     """
-    This is an entrypoint for the model training in metric learning setup.
+    This pipeline allow you to train and validate a feature extractor which
```
Comment (DaloroAT) [inline oml/lightning/pipelines/train.py:62]: allows

File: oml/lightning/pipelines/train_postprocessor.py [hunk 2]
```diff
@@ -111,6 +111,14 @@ def get_loaders_with_embeddings(cfg: TCfg) -> Tuple[DataLoader, DataLoader]:
 
 
 def postprocessor_training_pipeline(cfg: DictConfig) -> None:
+    """
+    This pipeline allow you to train and validate a pairwise postprocessor
```
Comment (DaloroAT) [inline oml/lightning/pipelines/train_postprocessor.py:115]: allows

File: oml/lightning/pipelines/validate.py [hunk 3]
```diff
@@ -22,10 +22,11 @@
 
 def extractor_validation_pipeline(cfg: TCfg) -> Tuple[pl.Trainer, Dict[str, Any]]:
     """
-    This is an entrypoint for the model validation in metric learning setup.
+    This pipeline allow you to validate a feature extractor which
```
Comment (DaloroAT) [inline oml/lightning/pipelines/validate.py:25]: allows

**Review by DaloroAT — APPROVED** [review]

**Pull request merged**